src/java.base/share/classes/sun/security/util/Password.java

```diff
@@ -51,13 +52,15 @@ public static char[] readPassword(InputStream in, boolean isEchoOn)
         byte[] consoleBytes = null;
 
         try {
-            // Use the new java.io.Console class
+            // Only use Console if `in` is the initial System.in
             Console con;
-            if (!isEchoOn && in == System.in && ((con = System.console()) != null)) {
+            if (!isEchoOn &&
+                    in == SharedSecrets.getJavaLangAccess().initialSystemIn() &&
+                    ((con = System.console()) != null)) {
                 consoleEntered = con.readPassword();
-                // readPassword returns "" if you just print ENTER,
+                // readPassword returns "" if you just press ENTER with the built-in Console,
                 // to be compatible with old Password class, change to null
-                if (consoleEntered != null && consoleEntered.length == 0) {
+                if (consoleEntered == null || consoleEntered.length == 0) {
```

Comment:
@wangweij Would you have cycles to build with this change to see that keytool is okay?

Reply:
I confirmed that the standalone `keytool` did not echo the input, which should be OK for this IMO.

Reply:
Naoto has confirmed that the password prompt from keytool does not echo, good!

The intention is that Console be usable in jshell so I think the issue is that readPassword is echo'ing when used in jshell. Maybe someone experimenting with the Console API might run into this but we can separate out that issue.

Reply:
I thought that in `jshell`, `System.console()` returns null, so it is always not using `Console.readPassword`. Anyway, I think the scenario is not practical (changing password using jshell), and in the future, jshell can provide its own Console implementation (that's a part of this enhancement's motivation) that would nicely handle this situation.